openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java

```diff
@@ -2388,6 +2446,41 @@ private void processPropertyElements(OpenAPI openAPI, VisitorContext context, El
         }
     }
 
+    private boolean notAllowedByJsonView(TypedElement publicField, List<String> classLvlJsonViewClasses, ClassElement jsonViewClassEl) {
+        List<String> fieldJsonViewClasses = ElementUtils.getJsonViewClasses(publicField);
+        if (CollectionUtils.isEmpty(fieldJsonViewClasses)) {
+            fieldJsonViewClasses = classLvlJsonViewClasses;
+        }
+        return CollectionUtils.isNotEmpty(fieldJsonViewClasses) && !checkJsonView(fieldJsonViewClasses, jsonViewClassEl);
+    }
+
+    private boolean checkJsonView(List<String> fieldJsonViewClasses, ClassElement jsonViewClassEl) {
+
+        if (fieldJsonViewClasses.contains(jsonViewClassEl.getName())) {
+            return true;
+        }
+
+        if (jsonViewClassEl.isInterface()) {
+            Collection<ClassElement> superInterfaces = jsonViewClassEl.getInterfaces();
+            if (CollectionUtils.isEmpty(superInterfaces)) {
+                return false;
+            }
+            for (ClassElement superInterface : superInterfaces) {
+                boolean found = checkJsonView(fieldJsonViewClasses, superInterface);
+                if (found) {
+                    return true;
+                }
+            }
+        } else {
+            ClassElement superType = jsonViewClassEl.getSuperType().orElse(null);
+            if (superType != null) {
+                return checkJsonView(fieldJsonViewClasses, superType);
+            }
+        }
+
+        return false;
```

Comment:
You can probably just call `visitContext.getClassElement(fieldJsonViewClassName)` to get the JsonViewClass and then just use `isAssignable` instead of manually traversing the class hierarchy? Seems like you have just reimplemented an assignment check

Reply:
fixed